src/app/Http/Controllers/SPPController.php

```diff
@@ -12,23 +14,39 @@ class SPPController extends Controller
      * Sending all active spp symbols
      */
     public function manage() {
-        return view();
+        return view('spp.manage', ['spp_symbols' => SppSymbol::where('status', SppStatus::ACTIVE)->pluck('spp_symbol', 'id')]);
     }
 
     /**
      * Same as store() in BusinessTrip
-     *Validating data
+     * Validating data
      * Saving new spp symbol in DB
      */
     public function store(Request $request) {
+        $validatedData = $request->validate([
+            'fund' => 'required|string',
+            'spp_symbol' => 'required|string|unique:spp_symbols,spp_symbol',
+            'functional_region' => 'required|string',
+            'account' => 'required|string',
+            'financial_centre' => 'required|string',
+            'grantee' => 'required|string|max:200',
+            ]);
+
+        SppSymbol::create($validatedData);
+
+        return redirect()->route('spp.manage')->with('message', 'ŠPP prvok bol pridaný.');
 
     }
 
     /**
      * Input parameter of type Request and the spp symbol
      * Updating the spp symbols state to deactivated
      */
-    public function deactivate(Request $request, Spp $spp) {
+//    public function deactivate(Request $request, SppSymbol $spp) {
+    public function deactivate(Request $request) {
+        $validatedData = $request->validate(['spp' => 'exists:spp_symbols,id']);
+        SppSymbol::find($validatedData['spp'])->update(['status' => SppStatus::DEACTIVATED]);
```

Comment:
Tu nemoze nastat null pointer exception? Alebo nam to osetruje ten validate vlastne?

Reply:
podla mna to urcite osetruje validate. lebo ak by sa tam nejako vyskytlo id spp prvku, ktory nie je, uz sa to nezvaliduje, teda vo formulari sa ukaze error pouzivatelovi. ale zaroven ani by nemalo byt mozne poslat form s takym id, ktore neexistuje, jedine, ze by ziadne aktivne spp prvky prave neboli. tak dropdown bude prazdny. takze myslim, ze tam pridam required a malo by to byt nepriestrelne.

Reply:
Super, vdaka